src/J2N/Collections/Arrays.cs

```diff
@@ -390,5 +392,224 @@ private static class EmptyArrayHolder<T>
+        /// to match Java's behavior with respect to NaN and signed zero comparisons.
+        /// </remarks>
+        /// <typeparam name="T">The type of the array element.</typeparam>
+        /// <param name="array">The array to search.</param>
+        /// <param name="value">The value to search for.</param>
+        /// <param name="startIndex">The zero-based starting index of the search.</param>
+        /// <returns>The zero-based index of the first occurrence of <paramref name="value"/>, if found; otherwise, -1.</returns>
+        public static int IndexOf<T>(T[] array, T value, int startIndex)
+        {
+            if (array is null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            if ((uint)startIndex > (uint)array.Length)
+                ThrowHelper.ThrowArgumentOutOfRange_IndexMustBeLessOrEqualException(startIndex);
+            return IndexOf(array, value, startIndex, array.Length - startIndex);
+        }
+
+        /// <summary>
+        /// Searches for the specified object in the array, starting from the specified index and searching up to the specified count,
+        /// and returns the index of the first occurrence.
+        /// </summary>
+        /// <remarks>
+        /// This method applies special-case handling for floating-point types (float, double, and their nullable variants)
+        /// to match Java's behavior with respect to NaN and signed zero comparisons.
+        /// </remarks>
+        /// <typeparam name="T">The type of the array element.</typeparam>
+        /// <param name="array">The array to search.</param>
+        /// <param name="value">The value to search for.</param>
+        /// <param name="startIndex">The zero-based starting index of the search.</param>
+        /// <param name="count">The number of elements in the section to search.</param>
+        /// <returns>The zero-based index of the first occurrence of <paramref name="value"/>, if found; otherwise, -1.</returns>
+        public static int IndexOf<T>(T[] array, T value, int startIndex, int count)
+        {
+            if (array is null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            if ((uint)startIndex > (uint)array.Length)
+                ThrowHelper.ThrowArgumentOutOfRange_IndexMustBeLessOrEqualException(startIndex);
+            if (count < 0 || startIndex > array.Length - count)
+                ThrowHelper.ThrowCountArgumentOutOfRange_ArgumentOutOfRange_Count(count);
+
+            // Special handling for float and double types to match Java's behavior
+            if (typeof(T) == typeof(float))
+            {
+                float floatValue = Unsafe.As<T, float>(ref value);
+                float[] floatArray = Unsafe.As<T[], float[]>(ref array);
+                return IndexOfFloatingPoint(floatArray, floatValue, startIndex, count, JCG.EqualityComparer<float>.Default);
```

Comment:
I probably would have went with using `ref T` instead of `T[]` here and brought over the support from [`SpanHelpers`](https://github.com/dotnet/runtime/blob/v10.0.0-rc.2.25502.107/src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs#L1480), but being that this is such an edge case that we will probably never encounter it, I think this is fine.